sdk/program/src/sysvar/stake_history.rs

```diff
@@ -58,9 +63,78 @@ impl Sysvar for StakeHistory {
     }
 }
 
+// we do not provide Default because this requires the real current epoch
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct StakeHistorySysvar(Epoch);
+
+impl StakeHistorySysvar {
+    pub fn new(current_epoch: Epoch) -> Result<Self, ProgramError> {
+        if current_epoch > 0 {
+            Ok(Self(current_epoch))
+        } else {
+            Err(ProgramError::InvalidAccountData)
+        }
+    }
+}
```

Comment:
Sorry if this is a dumb question or I'm missing something obvious, but how does this work for testing? With something like program-test, do we always need to advance the clock for it to work? Or if I'm using solana-test-validator, will the stake program fail during the first epoch?

I see that we avoid using `get_entry` for things like `current_epoch == target_epoch`, so does that help us avoid ever touching the error path?

Reply:
this is because i thought the first epoch was 1 rather than 0... ill remove it and see how it affects the math